Service/Sales/Quote/Recreate.php

```diff
@@ -106,20 +106,41 @@ public function recreate($order = false, $newQuote = false)
     public function recreateById($quoteId)
     {
         try {
-            $quote = $this->quoteFactory->create()->load($quoteId);
+            $oldQuote = $this->quoteFactory->create()->load($quoteId);
         } catch (\Exception $e) {
             $this->messageManager->addErrorMessage($e->getMessage());
         }
-        if ($quote->getId()) {
+
+        if ($oldQuote->getId()) {
             $emptyQuoteId = $this->quoteManagement->createEmptyCart();
-            $newQuote = $this->quoteFactory->create()->load($emptyQuoteId);
-            
-            $this->cart->setQuote($newQuote)->save();
-            $this->cart->saveQuote();
-            $this->checkoutSession->setQuoteId($newQuote->getId());
-            $this->checkoutSession->getQuote()->collectTotals()->save();
+            $quote = $this->quoteFactory->create()->load($emptyQuoteId);
+
+            $quote->merge($oldQuote)->save();
```

Comment:
can this fail? if so, can we catch and log an exception with details in case this fails?

Reply:
sure possible, exception added, Thank you